presto-orc/src/main/java/com/facebook/presto/orc/OrcWriter.java

```diff
@@ -246,6 +252,7 @@ public OrcWriter(
         this.metadataWriter = new CompressedMetadataWriter(orcEncoding.createMetadataWriter(), columnWriterOptions, Optional.empty());
         this.hiveStorageTimeZone = requireNonNull(hiveStorageTimeZone, "hiveStorageTimeZone is null");
         this.stats = requireNonNull(stats, "stats is null");
+        this.streamSizeHelper = new StreamSizeHelper(orcTypes, columnWriterOptions.getFlattenedNodes(), columnWriterOptions.isMapStatisticsEnabled());
```

Comment:
StreamSizeHelper is responsible for accumulating storage sizes for nodes and flat map keys.